src/audio/selector/selector.c

```diff
@@ -618,13 +620,32 @@ static int selector_init(struct processing_module *mod)
 
 	md->private = cd;
 
-	ret = memcpy_s(&cd->output_format, sizeof(cd->output_format),
-		       base_cfg + 1, sizeof(struct ipc4_audio_format));
+	expected_size = sizeof(struct ipc4_base_module_cfg);
+	expected_size += ipc4_calc_base_module_cfg_ext_size(SEL_NUM_IN_PIN_FMTS,
+							    SEL_NUM_OUT_PIN_FMTS);
+	if (cfg->size != expected_size) {
+		comp_err(mod->dev, "selector_init(): Invalid configuration size");
+		goto fail;
+	}
+
+	if (init_cfg->base_cfg_ext.nb_input_pins != SEL_NUM_IN_PIN_FMTS ||
+	    init_cfg->base_cfg_ext.nb_output_pins != SEL_NUM_OUT_PIN_FMTS) {
+		comp_err(mod->dev, "selector_init(): Invalid pin configuration");
+		goto fail;
+	}
```

Comment:
can you move all these checks above allocation, then you don't need to free in case of a failure